source/main/webpack.config.js

```diff
@@ -1,106 +1,105 @@
-   */
-  node: {
-    __dirname: false,
-    __filename: false,
-  },
-  externals: {
-    'js-chain-libs-node': 'commonjs2 js-chain-libs-node',
-    'trezor-connect': 'commonjs2 trezor-connect',
+  devtool: isDevelopment ? 'eval-source-map' : 'source-map',
+  optimization: {
+    minimize: false,
   },
   resolve: {
-    extensions: ['.tsx', '.ts', '.js', '.json'],
+    symlinks: true, // for native libraries
+    extensions: ['.ts', '.tsx', '.js', '.json'],
   },
   module: {
     rules: [
       {
         test: /\.tsx?$/,
         include: /source/,
         exclude: /source\/renderer/,
-        use: (isCi ? [] : ['cache-loader']).concat([
-          {
-            loader: 'babel-loader',
-            options: {
-              presets: [
-                '@babel/preset-env',
-                '@babel/preset-react',
-                '@babel/preset-typescript',
-              ],
+        loader: 'swc-loader',
+        options: {
+          jsc: {
+            parser: {
+              syntax: 'typescript',
             },
+            target: 'es2019',
+            loose: false,
           },
-        ]),
-      },
-      {
-        test: /(pdfkit|linebreak|fontkit|unicode|brotli|png-js).*\.js$/,
-        use: {
-          loader: 'transform-loader?brfs',
         },
       },
       {
         test: /\.(woff2?|eot|ttf|otf|png|jpe?g|gif|svg)(\?.*)?$/,
         exclude: /\.inline\.svg$/,
-        use: {
-          loader: 'file-loader',
-          options: {
-            name: '[name]-[hash].[ext]',
-            outputPath: 'assets/',
-          },
-        },
-      },
-      {
-        test: /\.mjs$/,
-        include: /node_modules/,
-        type: 'javascript/auto',
+        type: 'asset/resource',
       },
     ],
   },
+  externalsPresets: { node: true }, // in order to ignore built-in modules like path, fs, etc.
+  externals: [
+    {
+      'js-chain-libs-node': 'commonjs2 js-chain-libs-node',
+      usb: 'commonjs2 usb',
+      'node-hid': 'commonjs2 node-hid',
+      'trezor-connect': 'commonjs2 trezor-connect',
+      pdfkit: 'commonjs2 pdfkit',
+      'usb-detection': 'commonjs2 usb-detection',
+    },
+  ],
   plugins: [
     new webpack.DefinePlugin(
       Object.assign(
-        {
-          'process.env.API_VERSION': JSON.stringify(
-            process.env.API_VERSION || 'dev'
-          ),
-          'process.env.NETWORK': JSON.stringify(
-            process.env.NETWORK || 'development'
-          ),
-          'process.env.MOCK_TOKEN_METADATA_SERVER_PORT':
-            process.env.MOCK_TOKEN_METADATA_SERVER_PORT || 0,
-          'process.env.MOBX_DEV_TOOLS': process.env.MOBX_DEV_TOOLS || 0,
-          'process.env.BUILD_NUMBER': JSON.stringify(
-            process.env.BUILD_NUMBER || 'dev'
-          ),
-          'process.env.IS_WATCH_MODE': process.env.IS_WATCH_MODE === 'true',
-          'process.env.KEEP_LOCAL_CLUSTER_RUNNING':
-            process.env.KEEP_LOCAL_CLUSTER_RUNNING === 'true',
-        },
+        {},
```

Comment:
It seems this plugin is no longer needed, should we remove it?

Reply:
yes probably 👍 would you like to setup a Jira ticket for this?